### PR TITLE
Don't render ad slots on Opera Mini Extreme mode

### LINKED ISSUE #7517: Don't render ad slots on the Opera Mini browser in Extreme mode
**Is your feature request related to a problem? Please describe.**
Opera Mini Extreme mode works by rendering the page on an intermediate server, executing client javascript for a small window of time (clicking on buttons etc, and rendering before and after states), and then returning a snapshot of this to the user.

Testing a page with ads on from inside the UK resulted in a page that looks like this:

Screenshot_20200816-121324

The space is reserved for an ad, but as ads use client side javascript, they have not been loaded (and may also be blocked by Opera Mini?).

The intermediate server that was used to render the Simorgh page in this case must be outside of the UK, even though the user is the UK, as we do not render the ad slot if the user is within the UK.

**Describe the solution you'd like**
We have already implemented an opera mini check here: https://github.com/bbc/simorgh/blob/latest/src/app/containers/Header/index.jsx#L32

We can use the same logic to ensure we don't render an ad slot on Opera Mini.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves https://github.com/bbc/simorgh/issues/7517

**Overall change:**
Don't render ad slots on Opera Mini Extreme Mode to avoid:

1. People from **inside** the UK to see the ad slot due to the intermediate server that Opera Mini Extreme Mode uses being **outside** of the UK

2. People from **outside** the UK can't see the ad due to this intermediate server being **inside** of the UK

**Code changes:**
- Create a custom hook to check if on Opera Mini Extreme Mode
- Add opera mini check in the Canonical Ad container to not render the ad slot
- Update Header container to use new custom hook

**Testing notes**

This will have to be double-checked on the test environment because we can't set the `bbc-adverts` header locally in Opera Mini to confirm that when this is set, the slot is not rendered with this new `isOperaMini` condition.

However, you can remove the `!showAdsBasedOnLocation` from the following condition, to be able to test it locally:
```
  if (!showAdsBasedOnLocation || isOperaMini) {
    return null;
  }
 ⬇️
  if (isOperaMini) {
    return null;
  }
```

---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [X] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/Ad/Canonical/index.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { ServiceContext } from '#contexts/ServiceContext';
 import { RequestContext } from '#contexts/RequestContext';
 import isLive from '#lib/utilities/isLive';
 import getAdsAriaLabel from '../utilities/getAdsAriaLabel';
+import OperaMiniDetection from '#hooks/OperaMiniDetection';
```
Comment (tochwill) [inline src/app/containers/Ad/Canonical/index.jsx]: Could you call this `useOperaMiniDetection` so it's obvious it's a hook?

**Review by tochwill — COMMENTED** [review]
Looks good, just one comment from me :)

**Review by tochwill — APPROVED** [review]

**Review by itimirichard — APPROVED** [review]
LGTM!

**Comment by PriyaKR** [comment]
Couldn't replicate the issue on test environment.But as discussed with @tochwill understood its safer to have this check.
Also regression looks good.Can be merged.

**Pull request merged**